src/backend/base/langflow/components/models/embedding_model.py

```
@@ -77,37 +78,53 @@
+                raise ValueError(f"Failed to load SentenceTransformer model: {e}") from e
+
+            class LangflowEmbeddingWrapper:
+                def embed_documents(self, texts: list[str]) -> list[list[float]]:
+                    return st_model.encode(texts, convert_to_numpy=True).tolist()
+
+                def embed_query(self, text: str) -> list[float]:
+                    return st_model.encode(text, convert_to_numpy=True).tolist()
+
+            return LangflowEmbeddingWrapper()
+
+        raise ValueError(f"Unknown provider: {provider}")
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)